pgns/127489.js

```diff
@@ -136,3 +138,156 @@ module.exports = [
+  },
+  {
+    node: 'notifications.propulsion.%s.chargeIndicator',
+    message: '%s Engine Charge Indicator'
+  },
+  {
+    node: 'notifications.propulsion.%s.preheatIndicator',
+    message: '%s Preheat Indicator'
+  },
+  {
+    node: 'notifications.propulsion.%s.highBoostPressure',
+    message: '%s Engine High Boost Pressure'
+  },
+  {
+    node: 'notifications.propulsion.%s.revLimitExceeded',
+    message: '%s Engine Rev Limit Exceeded'
+  },
+  {
+    node: 'notifications.propulsion.%s.eGRSystem',
+    message: '%s Engine EGR System'
+  },
+  {
+    node: 'notifications.propulsion.%s.throttlePositionSensor',
+    message: '%s Engine Throttle Position Sensor'
+  },
+  {
+    node: 'notifications.propulsion.%s.emergencyStopMode',
+    message: '%s Engine Emergency Stop Mode'
+  }
+];
+
+var status2Notifications = [
+  { 
+    node: 'notifications.propulsion.%s.warningLevel1',
+    message: '%s Engine Warning Level 1'
+  },
+  {
+    node: 'notifications.propulsion.%s.warningLevel2',
+    message: '%s Engine Warning Level 2'
+  },
+  {
+    node: 'notifications.propulsion.%s.powerReduction',
+    message: '%s Engine Power Reduction'
+  },
+  {
+    node: 'notifications.propulsion.%s.maintenanceNeeded',
+    message: '%s Engine Maintenance Needed'
+  },
+  {
+    node: 'notifications.propulsion.%s.commError',
+    message: '%s Engine Comm Error'
+  },
+  {
+    node: 'notifications.propulsion.%s.subOrSecondaryThrottle',
+    message: '%s Engine Sub or Secondary Throttle'
+  },
+  {
+    node: 'notifications.propulsion.%s.neutralStartProtect',
+    message: '%s Neutral Start Protect'
+  },
+  {
+    node: 'notifications.propulsion.%s.shuttingDown',
+    message: '%s Engine Shutting Down'
+  }
+];
+
+
+function generateMappingsForStatus(field, notifications)
+{
+  notifications.forEach((notif,index) => {
+    var mapping = {
+      node: function(n2k) {
+        var engine = n2k.fields['Engine Instance'] === 'Single Engine or Dual Engine Port' ? "port" : "starboard";
+        return util.format(notif.node, engine);
+      },
+      filter: function(n2k) { return typeof n2k.fields[field] !== 'undefined'; },
+      value: function(n2k, state) {
+        var engine = n2k.fields['Engine Instance'] === 'Single Engine or Dual Engine Port' ? "Port" : "Starboard";
+        
+        if ( n2k.fields[field] != 0 ) {
+          if ( n2k.fields[field] & (1<<index) ) {
+            if ( typeof state === 'undefined'
+                 || typeof state[notif.node] === 'undefined'
+                 || state[notif.node] != 'alarm') {
+              if ( typeof state !== 'undefined' )
+                state[notif.node] = 'alarm'
+              return {
+                state: 'alarm',
+                method: [ "visual", "sound" ],
+                message: util.format(notif.message, engine)
+              }
+            } else {
+              return null
+            }
+          }
+        } else if ( typeof state !== 'undefined'
```

Comment:
Guess how I caught it? By trying to figure out what the boolean clauses actually mean by giving them names...